src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -480,31 +515,102 @@ public override async Task RunAsync()
+                }
+                PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc6HostRuntime);
+                return true;
+            }
+            // Throw an exception if HostRuntime is set to none of the expected values
+            throw new CliException($"Invalid host runtime '{HostRuntime}'. Valid values are 'default', '{DotnetConstants.InProc8HostRuntime}', '{DotnetConstants.InProc6HostRuntime}'.");
+        }
+
+        private void PrintVerboseForHostSelection(bool isVerbose, string hostRuntime)
+        {
+            if (isVerbose)
+            {
+                ColoredConsole.WriteLine(VerboseColor($"Selected {hostRuntime} host"));
+            }
+        }
+
+        private bool ShouldLaunchChildProcessAfterInferringHostRuntimeAsync(bool shouldLaunchNet8, bool isInProc, bool isVerbose)
+        {
+            // Check if the app is in-proc
+            if (isInProc)
+            {
+                // Start .NET 8 child process if InProc8 is enabled
+                if (shouldLaunchNet8)
+                {
+                    PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc8HostRuntime);
+                }
+                // Otherwise start .NET 6 child process since we are running an inproc app
+                else
+                {
+                    PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc6HostRuntime);
+                }
+                return true;
+            }
+            PrintVerboseForHostSelection(isVerbose, "out-of-process");
+            return false;
+          
+        }
+
         private static string GetInProcNet8ExecutablePath()
         {
             var funcExecutableDirectory = Path.GetDirectoryName(typeof(StartHostAction).Assembly.Location)!;
-            var executableName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? WindowsExecutableName : LinuxExecutableName;
+            var executableName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? DotnetConstants.WindowsExecutableName : DotnetConstants.LinuxExecutableName;
 
-            return Path.Combine(funcExecutableDirectory, InProc8DirectoryName, executableName);
+            return Path.Combine(funcExecutableDirectory, DotnetConstants.InProc8DirectoryName, executableName);
         }
 
-        private Task StartInProc8AsChildProcessAsync()
+        private static string GetInProcNet6ExecutablePath()
+        {
+            var funcExecutableDirectory = Path.GetDirectoryName(typeof(StartHostAction).Assembly.Location)!;
+            var executableName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? DotnetConstants.WindowsExecutableName : DotnetConstants.LinuxExecutableName;
+
+            return Path.Combine(funcExecutableDirectory, DotnetConstants.InProc6DirectoryName, executableName);
+        }
+
+        private async Task StartHostAsChildProcessAsync(bool shouldStartNet8ChildProcess)
```

Comment:
Nothing is being awaited on, we can remove `async` here.

Reply:
removed